### PR TITLE
Desktop Window refactor

### PR DESCRIPTION
**Pull request by chicoxyzzy**
It's not ready for review yet. I want to test things on CI (at some point) and let people see what upcoming changes will be (see `test/application.js` and `test/window/js`)

Tests could (and probably will) now as I run `ssc build` for now and don't care about the standard linter most of the time so I can see the Web Inspector logs

### CODE REVIEW

File: api/application.js [hunk 1]
```diff
@@ -0,0 +1,289 @@
...
+  const windows = await getWindows([index])
+  return windows[index]
+}
+
+/**
+ * Returns the ApplicationWindow instance for the current window.
+ * @return {Promise<ApplicationWindow>}
+ */
+export async function getCurrentWindow () {
+  return getWindow({ index: globalThis.__args.index })
```
Comment (jwerle) [inline api/application.js:133]: this should be `  return getWindow(globalThis.__args.index)`

File: api/application.js [hunk 2]
```diff
@@ -0,0 +1,303 @@
+/**
+ * @module Application
```
Comment (heapwolf) [inline api/application.js:2]: maybe we want to shorten this to `app`?

File: src/desktop/main.cc [hunk 3]
```diff
@@ -652,72 +567,159 @@ MAIN {
...
+
+      JSON::Object json = JSON::Object::Entries {
+        { "data", targetWindow->json() },
+      };
+
+      currentWindow->resolvePromise(seq, OK_STATE, json.str());
+      return;
+    }
+
+    if (message.name == "window.showInspector") {
```
Comment (heapwolf) [inline src/desktop/main.cc:709]: does this work on linux and windows? cc @mribbons

**Review by heapwolf — APPROVED** [review]
this work looks great, from what I read it looks good to merge and start trying to use it.

File: src/desktop/main.cc (hunk 3 above)
Comment (mribbons) [reply src/desktop/main.cc:709]: Inspector menu works on windows :+1: @chicoxyzzy @heapwolf

**Review by mribbons — APPROVED** [review]

File: src/core/core.cc [hunk 4]
```diff
@@ -587,14 +587,14 @@ namespace SSC {
 
       if (!success) {
         json = JSON::Object::Entries {
-          {"source", "platform.notify"},
```
Comment (chicoxyzzy) [inline src/core/core.cc:590]: oh, wasn't it for the OS notifications?

**Review by chicoxyzzy — COMMENTED** [review]
👍

**Review by jwerle — APPROVED** [review]

File: src/core/core.cc (hunk 4 above)
Comment (jwerle) [reply src/core/core.cc:590]: Not there. It was a typo
